trinity/utils/datastructures.py

```diff
@@ -36,6 +38,56 @@ def __set__(self, oself: Any, value: TFunc) -> None:
         self._func = value
 
 
+@total_ordering
+class SortableTask(Generic[TTask]):
+    _order_fn: FunctionProperty[Callable[[TTask], Any]] = None
```

Comment:
Yeah, actually maybe a better name for it would be `StaticMethod` since that's really the problem it's circumventing: mypy expects it to be bound and receive `self` as the first arg with out this hack.